presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java

```diff
@@ -82,14 +83,25 @@ public CachingHiveMetastore(@ForCachingHiveMetastore ExtendedHiveMetastore deleg
 
     public CachingHiveMetastore(ExtendedHiveMetastore delegate, ExecutorService executor, Duration cacheTtl, Duration refreshInterval)
     {
-        this.delegate = requireNonNull(delegate, "delegate is null");
+        this(requireNonNull(delegate, "delegate is null"),
+                requireNonNull(executor, "executor is null"),
+                OptionalLong.of(requireNonNull(cacheTtl, "cacheTtl is null").toMillis()),
+                OptionalLong.of(requireNonNull(refreshInterval, "refreshInterval is null").toMillis()));
+    }
 
-        long expiresAfterWriteMillis = requireNonNull(cacheTtl, "cacheTtl is null").toMillis();
-        long refreshMills = requireNonNull(refreshInterval, "refreshInterval is null").toMillis();
+    public static CachingHiveMetastore memoizeMetastore(ExtendedHiveMetastore delegate, ExecutorService executor)
+    {
+        return new CachingHiveMetastore(requireNonNull(delegate, "delegate is null"),
+                requireNonNull(executor, "executor is null"),
```

Comment:
I just realize we don't need to pass in an `executor` here. You can use `DirectExecutor`.